tools/render-test/shader-input-layout.cpp

```diff
@@ -42,596 +42,641 @@ namespace renderer_test
         return SLANG_SCALAR_TYPE_NONE;
     }
 
-    Index ShaderInputLayout::findEntryIndexByName(const String& name) const
+    void ShaderInputLayout::AggVal::addField(ShaderInputLayout::Field const& field)
     {
-        const Index count = Index(entries.getCount());
-        for (Index i = 0; i < count; ++i)
-        {
-            const auto& entry = entries[i];
-            if (entry.name == name)
-            {
-                return Index(i);
-            }
-        }
-        return -1;
+        fields.add(field);
     }
 
-    static bool _isCPULikeBindingTarget(SlangCompileTarget target)
+    void ShaderInputLayout::ArrayVal::addField(ShaderInputLayout::Field const& field)
     {
-        // CUDA and C++ are 'CPULike' in terms of their binding mechanism
-        switch (target)
+        vals.add(field.val);
+    }
+
+    class ShaderInputLayoutFormatException : public Exception
+    {
+    public:
+        ShaderInputLayoutFormatException(String message)
+            : Exception(message)
+        {}
+    };
+
+    struct ShaderInputLayoutParser
+    {
+        ShaderInputLayout* layout;
+        RandomGenerator* rand;
+
+        ShaderInputLayoutParser(ShaderInputLayout* layout, RandomGenerator* rand)
+            : layout(layout)
+            , rand(rand)
+        {}
+
+        RefPtr<ShaderInputLayout::ParentVal> parentVal;
+        List<RefPtr<ShaderInputLayout::ParentVal>> parentValStack;
+
+        SlangResult parseOption(TokenReader& parser, String const& word, ShaderInputLayout::TextureVal* val)
         {
-            case SLANG_C_SOURCE:
-            case SLANG_CPP_SOURCE:
-            case SLANG_EXECUTABLE:
-            case SLANG_SHARED_LIBRARY:
-            case SLANG_HOST_CALLABLE:
-            case SLANG_CUDA_SOURCE:
-            case SLANG_PTX:
+            if (word == "depth")
+            {
+                val->textureDesc.isDepthTexture = true;
+            }
+            else if (word == "arrayLength")
+            {
+                parser.Read("=");
+                val->textureDesc.arrayLength = parser.ReadInt();
+            }
+            else if (word == "size")
             {
-                return true;
+                parser.Read("=");
+                auto size = parser.ReadInt();
+                val->textureDesc.size = size;
             }
-            default: return false;
+            else if (word == "content")
+            {
+                parser.Read("=");
+                auto contentWord = parser.ReadWord();
+                if (contentWord == "zero")
```

Comment:
We may want to add support for arbitrary constant values so we have a way to test whether the right texture is being bound.